### ISSUE TITLE
Allow installing/activating multiple packages at once

### ISSUE THREAD

**Issue by raxod502** [raxod502/straight.el#286]
Once #76 is completed, or even just once we can clone packages asynchronously, it would be nice to let the user provide a list of all their recipes up-front so then everything can be done in parallel.

**Comment by Compro-Prasad** [comment]
Is there a way to start downloading next package while the current one is being compiled or is being processed. I wanted this feature because yasnippet takes so long to download and compile/extract.

**Comment by raxod502** [comment]
No, all operations are synchronous except for the experimental function `straight-x-fetch-all`.

**Comment by raxod502** [comment]
Yes, of course. That would be nice. I was simply saying that this is not implemented at present. See https://github.com/raxod502/straight.el/issues/103.

Note also that Elisp does not do real multithreading. The only parallelism in Elisp is by spawning external processes in the background.

**Comment by sooheon** [comment]
Reference for a potential UI for this operation that makes for a nice demo, imo:
![](https://raw.githubusercontent.com/junegunn/i/master/vim-plug/installer.gif)

**Comment by raxod502** [comment]
The extension command `straight-x-fetch-all` implements this interface for `straight.el`.
